pkg/controller/storagecluster/external_resources.go

```diff
@@ -77,15 +76,86 @@ func (r *ReconcileStorageCluster) ensureExternalStorageClusterResources(instance
 	}
 	err := r.client.Get(context.TODO(), types.NamespacedName{Name: found.Name, Namespace: found.Namespace}, found)
 	if err != nil {
-		return err
+		reqLogger.Error(err, "could not find the external secret resource")
+		return nil, err
 	}
 	var data []ExternalResource
 	err = json.Unmarshal(found.Data[externalClusterDetailsKey], &data)
 	if err != nil {
 		reqLogger.Error(err, "could not parse json blob")
-		return err
+		return nil, err
+	}
+	return data, nil
+}
+
+func newExternalGatewaySpec(rgwEndpoint string, reqLogger logr.Logger) (*cephv1.GatewaySpec, error) {
+	var gateWay cephv1.GatewaySpec
+	hostIP, portStr, err := net.SplitHostPort(rgwEndpoint)
+	if err != nil {
+		reqLogger.Error(err,
+			fmt.Sprintf("invalid rgw endpoint provided: %s", rgwEndpoint))
+		return nil, err
+	}
+	if hostIP == "" {
+		err := fmt.Errorf("An empty rgw host 'IP' address found")
+		reqLogger.Error(err, "Host IP should not be empty in rgw endpoint")
+		return nil, err
+	}
+	gateWay.ExternalRgwEndpoints = []corev1.EndpointAddress{{IP: hostIP}}
+	if portInt64, err := strconv.ParseInt(portStr, 10, 32); err != nil {
```

Comment:
```suggestion
	portInt64, err := strconv.ParseInt(portStr, 10, 32)
	if err != nil {
		reqLogger.Error(err,
			fmt.Sprintf("invalid rgw 'port' provided: %s", portStr))
		return nil, err
	}
	gateWay.Port = int32(portInt64)
```

Reply:
Can you please refractor it a little? It's giving lint error

Reply:
Corrected...